### DISCUSSION TITLE
What should I do after creating a new folder?

### DISCUSSION THREAD

**Discussion by benyltam** [community/community#34800] [Other Feature Feedback, Questions, & Ideas] [answered]
I create a folder on the repository window by choosing "Add file > Create new file > [New folder]/tmp.txt" and then delete the file tmp.txt. Then I "commit" the change. However, when I go back to the repository folder, I don't see the new folder. I wonder if I should do something after "committing" the change. Is there anybody who knows what should I do?

**Comment by airtower-luna** [comment, marked as answer]
Git doesn't handle empty folders at all, it only considers them as parents of tracked files. So you cannot commit an empty folder.

If you absolutely need the folder to exist within the repository, you have to commit a file inside it (the file can be empty). In many cases it'll be better to create the folder as part of an installation procedure, but that depends on the code in your repository.

**Reply by benyltam** [reply]
Thanks for your reply!

**Comment by mikehcox** [comment]
This is a "feature" of git. All directories must have at least one file in them. Git tracks the contents/versions of *files* not directories. If you want to create an "empty" directory, put an empty .gitignore file in it.

**Comment by ht-vo** [comment]
When I need to keep an empty directory, I simply create an empty file named `.gitkeep`.
